cou/apps/auxiliary.py

```diff
@@ -324,12 +324,21 @@ def post_upgrade_steps(
         return steps
 
     def _check_auto_restarts(self) -> None:
-        """No-op, skip check auto restarts option.
+        """Check if enable-auto-restarts is enabled.
 
-        This method override the parent class's `_check_auto_restarts()` method
-        because the parent class's will raise an `ApplicationError` if
-        `enable-auto-restarts` is `True`.
+        Due to the charm [bug](https://bugs.launchpad.net/charm-rabbitmq-server/+bug/2046381),
+        if the `enable-auto-restarts` option is enabled, this check will raise an exception.
+
+        :raises ApplicationError: When enable-auto-restarts is enabled.
         """
+        if self.config["enable-auto-restarts"].get("value") is True:
+            raise ApplicationError(
+                "`enable-auto-restarts` must be `False` due to "
+                "https://bugs.launchpad.net/charm-rabbitmq-server/+bug/2046381 "
+                f"Please run `juju config {self.name} enable-auto-restarts=False` "
+                "before performing upgrades and rollback to original value after "
+                "upgrade is completed"
```

Comment:
I think it would be a better user experience to show a message showing that is a know bug and recommend to change the configuration and prompt if the user wants to proceed instead of raising ApplicationError.

Not sure if that would require a lot of refactor or increase the complexity...

Reply:
@gabrielcocenza there aren't many things we prompt for like this - I think only vault upgrade? Could you open a github issue about it to consider more prompts in the design of COU (and how this should work with no-prompt mode)?

Something related I realised that would improve the ux here: this would be a good candidate for a pre-plan check in verify_cloud. We could keep both: a hard error here, but also a verify-cloud check to alert the user that they need to change the config before they even start upgrading (along with the rest of the warnings about the cloud). What do you think @chanchiwai-ray ?

Reply:
I vaguely remember that pre_plan_sanity_check will also run before upgrade, that's why I can still put it here. 

>Something related I realised that would improve the ux here: this would be a good candidate for a pre-plan check in verify_cloud....

pre_plan_sanity_check and verify_cloud has similar functionality, and we should consider migrating pre_plan_sanity_check to verify_cloud

Reply:
> I think it would be a better user experience to show a message showing that is a know bug and recommend to change the configuration and prompt if the user wants to proceed instead of raising ApplicationError.

adding a prompt may make the `--auto-approve` and other logic more complex: do we auto approve in this situation? even though it's not working when it's `True`? do the user know you will change the `enable-auto-restarts=False` on behalf of them? How do you guarantee COU can rollback to the original setting in case of failure during the upgrade, etc... I think those are the complexities...

Reply:
Ah right my bad, I missed that this added check is part of the upgrade_plan_sanity_checks. So this is fine by me, and we can refactor later to merge the sanity checks with verify cloud. :+1: :+1: